.github/workflows/python-lint.yml

```diff
@@ -0,0 +1,46 @@
+name: lint PR
+on:
+  pull_request:
+    paths:
+      - .github/workflows/python-lint.yml
+      - .
+  push:
+    branches:
+      - main
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/main'}}
+jobs:
+  linter:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.10"]
+    steps:
+    - name: Git checkout
+      uses: actions/checkout@v3.3.0
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ matrix.python-version }}      
+    - name: Install linters
+      run: |
+        pip install black==22.* \
+          flake8 \
```

Comment:
It would be nice to have them as dependencies of the project and just do something like `poetry install --with linters` :)

Reply:
(Also, we should probably pin flake8 and mypy)

Reply:
lock version, done